script/EstimateMaxCount.s.sol

```diff
@@ -0,0 +1,57 @@
+// solhint-disable no-console
+pragma solidity >=0.8.22 <0.9.0;
+
+import { LockupDynamic_Gas_Test } from "test/benchmark/LockupDynamic.Gas.t.sol";
+import { LockupTranched_Gas_Test } from "test/benchmark/LockupTranched.Gas.t.sol";
+
+import { BaseScript } from "./Base.s.sol";
+
+contract EstimateMaxCount is BaseScript {
+    /// @notice Estimate the maximum number of segments
+    /// @param blockGasLimit The block gas limit
+    /// @return count maximum number of segments that can be created within the block gas limit
+    /// @return gasUsed gas consumed by the function when maximum number of segments are created
+    function estimateSegments(uint256 blockGasLimit) public virtual returns (uint128 count, uint256 gasUsed) {
+        count = 240;
+
+        LockupDynamic_Gas_Test lockupDynamicGasTest = new LockupDynamic_Gas_Test();
+        lockupDynamicGasTest.setUp();
+
+        uint256 gasConsumed = 0;
+        while (blockGasLimit > gasConsumed + 21_000) {
```

Comment:
Agree with your point. I will add `BASE_FEE` for each chain but it will still not be accurate (see [my comment](https://github.com/sablier-labs/v2-core/pull/908#issuecomment-2098511678)). 

I am also thinking of renaming it to `ETH_TRANSFER_GAS`. `BASE_FEE` is defined as the base cost of 1 unit of gas  in wei.

Reply:
As to mitigate the risk because of dynamic gas fee in zkSync, how about we deduct 100 from the final count (for all chains so we don't have to hack pick it for each), until I have a robust mechanism to derive the actual execution cost of a tx? Re: https://twitter.com/sy100x/status/1787849733525913701

Reply:
Good idea to rename it to `ETH_TRANSFER_GAS`.

And fantastic idea to deduct 1M as a precaution.